odysseus/odysseus_tree/board/tpu/plugin_user_property_timestamp.c

```diff
@@ -0,0 +1,48 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/time.h>
+
+
+#include "nanomq/plugin.h"
+
+/*
+ * How to compile:
+ * gcc -I../ -fPIC -shared plugin_user_property.c -o plugin_user_property.so
+ */
+
+
+int cb(void *data)
+{
+	char **property = data;
+	if (property != NULL) {
+		
+		// key of ts, add 1 bc termination byte
+		property[0] = malloc(strlen("ts") + 1);
+		strcpy(property[0], "ts");
```

Comment:
i think you can just do this, or sprintf instead of mallocing first right

Reply:
Wouldn't it have a different length when it is char array versus when it is a long? dont rlly know what u mean ngl.